### PR TITLE
Latest `model-engine` updates

### PR DESCRIPTION
**Pull request by phil-scale**
Adds `model-engine` folder with all the code that runs the production serving for CRUD and sending tasks to model endpoints, async and sync.

### CODE REVIEW

**Review by song-william — APPROVED** [review]

File: model-engine/model_engine_server/common/constants.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from pathlib import Path
+
+BILLING_POST_INFERENCE_HOOK: str = "billing"
+CALLBACK_POST_INFERENCE_HOOK: str = "callback"
+READYZ_FPATH: str = "/tmp/readyz"
+DEFAULT_CELERY_TASK_NAME: str = "hosted_model_inference.inference.async_inference.tasks.predict"
+LIRA_CELERY_TASK_NAME: str = "ml_serve.celery_service.exec_func"
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/common/constants.py:7]: are these two still used?

Comment (phil-scale) [reply model-engine/model_engine_server/common/constants.py:7]: Yeah there are still legacy endpoints on the default task name

File: model-engine/model_engine_server/db/scripts/fivetran_setup/launch.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
+CREATE USER fivetran PASSWORD 'fivetran';
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/db/scripts/fivetran_setup/launch.sql:1]: are these new and do we want these here?

Comment (phil-scale) [reply model-engine/model_engine_server/db/scripts/fivetran_setup/launch.sql:1]: deleting

File: model-engine/model_engine_server/domain/use_cases/trigger_use_cases.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+from model_engine_server.domain.exceptions import CronSyntaxException, EndpointLabelsException
+from model_engine_server.domain.gateways.cron_job_gateway import CronJobGateway
+from model_engine_server.domain.repositories import (
+    DockerImageBatchJobBundleRepository,
+    DockerRepository,
+    TriggerRepository,
+)
+from model_engine_server.domain.use_cases.model_endpoint_use_cases import validate_labels
+
+DEFAULT_HOST = "https://model-engine.ml-serving-internal.scale.com"
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/domain/use_cases/trigger_use_cases.py:34]: nope

Comment (phil-scale) [reply model-engine/model_engine_server/domain/use_cases/trigger_use_cases.py]: oops sorry

File: model-engine/model_engine_server/inference/service_requests.py [hunk 4]
```diff
@@ -48,7 +48,7 @@ def get_s3_client():
 
 def _read_function_to_network_endpoint_info():
     # Dictionary format: {servable_id: {remote: true/false, endpoint_type: "sync"/"async", destination: <str>},...}
-    # destination is either a celery queue name, i.e. llm_engine_server.<something>, or the full url for an http request,
+    # destination is either a celery queue name, i.e. launch.<something>, or the full url for an http request,
```
Comment (yunfeng-scale) [inline model-engine/model_engine_server/inference/service_requests.py:51]: update some comments?

Comment (phil-scale) [reply model-engine/model_engine_server/inference/service_requests.py:51]: actually the celery queue name is still prefixed with `launch`

**Pull request merged**